nixos/tests/containers-tmpfs.nix

```diff
@@ -0,0 +1,45 @@
+# Test for NixOS' container support.
+
+import ./make-test.nix ({ pkgs, ...} : {
+  name = "containers-bridge";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ ckampka ];
+  };
+
+  machine =
+    { config, pkgs, ... }:
+    { imports = [ ../modules/installer/cd-dvd/channel.nix ];
+      virtualisation.writableStore = true;
+      virtualisation.memorySize = 768;
+
+      containers.webserver =
+        {
+          autoStart = true;
+          privateNetwork = true;
+          tmpfs = [ "/var" ];
+          config =
+            { services.httpd.enable = true;
+              services.httpd.adminAddr = "foo@example.org";
+              networking.firewall.allowedTCPPorts = [ 80 ];
+              networking.firewall.allowPing = true;
+            };
+
+        };
+
+      virtualisation.pathsInNixDB = [ pkgs.stdenv ];
+    };
+
+  testScript =
+    ''
+      $machine->waitForUnit("default.target");
+      $machine->succeed("nixos-container list") =~ /webserver/ or die;
+
+      # Start the webserver container.
+      $machine->succeed("nixos-container status webserver") =~ /up/ or die;
+
+      # Verify that /var is mounted as a tmpfs
+      $machine->succeed("nixos-container run webserver -- systemctl status var.mount --no-pager 2>/dev/null") =~ /What: tmpfs/ or die;
```

Comment:
Here you could extend the test to check that files placed in var in the container are not visible outside the container and the other way round (visible on the host but not in the container).